File: app/src/main/java/com/kirawii/thunderswufe/network/ElectricityService.java
package com.kirawii.thunderswufe.network;

import java.util.Map;
import retrofit2.Call;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface ElectricityService {
    @FormUrlEncoded
    @POST("easytong_app/GetPayAccInfoNew")
    Call<ElectricityResponse> getElectricityInfo(
        @Header("Authorization") String authorizationToken,
        @Header("User-Agent") String userAgent,
        @Header("Accept") String accept,
        @Header("Origin") String origin,
        @Header("Referer") String referer,
        @Header("h5req") String h5req,
        @Header("sec-ch-ua-platform") String secChUaPlatform,
        @Header("sec-ch-ua") String secChUa,
        @Header("sec-ch-ua-mobile") String secChUaMobile,
        @Header("x-requested-with") String xRequestedWith,
        @Header("sec-fetch-site") String secFetchSite,
        @Header("sec-fetch-mode") String secFetchMode,
        @Header("sec-fetch-dest") String secFetchDest,
        @Header("accept-language") String acceptLanguage,
        @FieldMap Map<String, String> requestFields
    );
} 

File: app/src/main/java/com/kirawii/thunderswufe/network/ElectricityService.kt
package com.kirawii.thunderswufe.network

import retrofit2.Response
import retrofit2.http.FieldMap
import retrofit2.http.FormUrlEncoded
import retrofit2.http.Header
import retrofit2.http.POST

/**
 * Retrofit service interface for interacting with the electricity API.
 */
interface ElectricityService {

    /**
     * Fetches electricity account information.
     * Assumes the API expects 'application/x-www-form-urlencoded' content type.
     */
    @FormUrlEncoded
    @POST("easytong_app/GetPayAccInfoNew") // Relative path to the endpoint
    suspend fun getElectricityInfo(
        @Header("Authorization") authorizationToken: String,
        @Header("User-Agent") userAgent: String, // As per Python script, passed explicitly
        @Header("Accept") accept: String = "application/json, text/plain, */*", // Default from Python
        @Header("Origin") origin: String, // As per Python script
        @Header("Referer") referer: String, // As per Python script
        // Cookies should be handled by OkHttp's CookieJar configured in NetworkModule

        // + Add new headers based on successful request log
        @Header("h5req") h5req: String = "Y",
        @Header("sec-ch-ua-platform") secChUaPlatform: String = "\"Android\"", // Encapsulate in escaped quotes
        @Header("sec-ch-ua") secChUa: String = "\"Android WebView\";v=\"135\", \"Not-A.Brand\";v=\"8\", \"Chromium\";v=\"135\"", // Encapsulate in escaped quotes
        @Header("sec-ch-ua-mobile") secChUaMobile: String = "?1",
        @Header("x-requested-with") xRequestedWith: String = "cn.com.yunma.school.app",
        @Header("sec-fetch-site") secFetchSite: String = "same-origin",
        @Header("sec-fetch-mode") secFetchMode: String = "cors",
        @Header("sec-fetch-dest") secFetchDest: String = "empty",
        @Header("accept-language") acceptLanguage: String = "zh-CN,zh;q=0.9,en-US;q=0.8,en;q=0.7",

        @FieldMap requestFields: Map<String, String> // All form fields including dynamic Time & Sign
    ): Response<ElectricityResponse>

    // TODO: Add other API endpoints here if needed
}